### PR TITLE
Apply /resources redesign

### PR DESCRIPTION
**Pull request by mtruj013**
## Done

- Applired redesign as per [doc](https://docs.google.com/document/d/1PB5-CamGmSjhbuyuGSariC9zUW-8krAJD0l8JyvNk20/edit) and [design](https://www.figma.com/file/0DTIS3uNNUuDAdCKWrk8Uf/23.10---Microk8s.io?type=design&node-id=0-1&mode=design&t=IiSkurvj9PifdsoF-0)
**NOTE: This page is missing the suru intentionally. I created a reusable class for it in a [separate PR](https://github.com/canonical/microk8s.io/pull/610) which is being reviewed. I will add that class to this hero once approved**
## QA

- View the site locally in your web browser at: https://microk8s-io-607.demos.haus/resources
- Check page against doc and design 

## Issue / Card

Fixes https://warthogs.atlassian.net/browse/WD-4192

### CODE REVIEW

File: static/sass/_pattern_resources.scss [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  display: flex;
+  flex-direction: column;
+  height: 338px;
+  max-width: 567px;
+  padding-left: 1rem;
+  padding-right: 1rem;
+  position: relative;
+
+  h2 {
+    color: white;
```
Comment (anthonydillon) [inline static/sass/_pattern_resources.scss:13]: Can you add `color` to the `.p-card--resources` element. If you set the background we should set color and we are not guaranteed to have h2 is all use cases.

File: templates/resources/videos-and-webinars.html [hunk 2]
```diff
@@ -0,0 +1,143 @@
+<hr class="p-rule">
+<h2 class="p-text--small-caps">Videos and webinars</h2>
+<article class="p-section--shallow">
+  <div class="p-card--resources">
+    <h2>What is MicroK8s?</h2>
+    <div class="icon-container">
+      <a href="https://www.youtube.com/watch?v=v9KI2BAF5QU"><i class="p-icon--video-play is-light"></i></a>
```
Comment (anthonydillon) [inline templates/resources/videos-and-webinars.html:7]:
```suggestion
      <a href="https://www.youtube.com/watch?v=v9KI2BAF5QU"><i class="p-icon--video-play is-light">Watch the What is MicroK8s? video</i></a>
```

Empty links are no accessible. Icon patterns already hide the content allowing linking to contain contextual text.

Comment (anthonydillon) [reply templates/resources/videos-and-webinars.html:7]: Same for all the media links in this doc

**Review by anthonydillon — CHANGES_REQUESTED** [review]
Looking good @mtruj013 some small suggestions in the code

**Comment by mtruj013** [comment]
Thanks @anthonydillon, could you take another look?

**Comment by juanruitina** [comment]
Please make the titles for all videos, publications and posts `h3` to keep a clean heading tree (formatting is OK). Other than that, LGTM.

**Review by anthonydillon — APPROVED** [review]
Perfect, thanks :+1:

**Comment by mtruj013** [comment]
@lyubomir-popov I made the changes, but I'm not sure I understood the spacing for the video and webinars section, so a few questions here if you don't mind

> shallow sections wrapping these images (one has shallow spacing, the other none):

Do you mean that shallow sections should wrap all but the last one? Or that the first one is supposed to be wrapped in a shallow section, but the rest are not? Also, in changing the wrapping divs for these to shallow sections instead shallow strips we lose the spacing between the hr and the card.
image

What is the spacing supposed to be for the separators? IIRC we're not supposed to be using utility classes and it's not specified in dev mode on the figma file. 

Lastly, [as Juan pointed out ](https://github.com/canonical/microk8s.io/pull/607#issuecomment-1684027009), the thumbnail overlaps with the text on small screens, what is the card supposed to look like in these situations?

Please let me know if the other changes I pushed are correct. Thanks!

**Comment by lyubomir-popov** [comment]
> And some visual feedback: the video thumbnail doesn't seem to downscale well on mobile, @lyubomir-popov any thoughts?

How about this design @juanruitina @mtruj013? I would make the whole card clickable. a Subtle affordance could be a gentle zoom on the background, as Ant did on the RFQ cards

image

**Comment by lyubomir-popov** [comment]
@mtruj013 sorry forgot to mention, everythign else looks great. One question about the sidenav - shouldn't it respod toscrolling and highlight the current section? ATM, if I click the last entry in it and relaod, it still stays highlighted, even though I'm at the top of the page - where the first sidenav item should be highlighted:
image

**Comment by mtruj013** [comment]
@lyubomir-popov I quite like that card design! I'll make the update and fix active tab change on scroll as well

**Comment by mtruj013** [comment]
@lyubomir-popov  Addressed everything with the exception of setting active tabs based on what's in the viewport. For context, this is a new use case for this component (that functionality doesn't exist on u.com or c.com). I had a go, but I'm still getting bugs and I don't have much more time to keep playing around with it. If you can think of an example where this is already live and implemented please link so I can take a look and replicate, but otherwise I think it might be better to make a follow up ticket

**Comment by lyubomir-popov** [comment]
Hi @mtruj013 sorry to keep pestering you, but there are still a few things to polish here:

- the second section should be a p-section--deep, not p-strip - a p-strip doubles the space between hero and the next thing. And also, since this section wraps all subsequent content, it should be deep - so once the dark footer comes in, there's a deep space (8rem) between the last item on paper background, and the dark, contrasting footer.
image

- the text zooms with the background - and it needs easing. ease-out, there's a preset in vanilla.

- the text here:
image
is still at a different distance from all others:
image


- this `hr` needs a `p-rule` class. Only `hr`s above images, buttons and other block elements should have no class. This is because they would otherwise stick to the element following them. All `hr`s that have a piece of text after them should have the `p-rule` class, as all text already has a padding top.

**Comment by lyubomir-popov** [comment]
> Addressed everything with the exception of setting active tabs based on what's in the viewport. For context, this is a new use case for this component (that functionality doesn't exist on u.com or c.com). I had a go, but I'm still getting bugs and I don't have much more time to keep playing around with it. If you can think of an example where this is already live and implemented please link so I can take a look and replicate, but otherwise I think it might be better to make a follow up ticket

@mtruj013 cc-ing @bartaz here who might know. I think we can launch like this and have a separate known issue to fix it later. Please see the comments above for a few small remaining glitches. Once those are addressed, this is good to go.

**Comment by lyubomir-popov** [comment]
video and text still diff widfths:
image

File: static/sass/_pattern_resources.scss [hunk 3]
```diff
@@ -33,3 +33,9 @@
     background: none !important;
   }
 }
+
+#from-our-blog img {
+  width:  35rem;
```
[the remark this answers is not in the document]
Comment (mtruj013) [reply static/sass/_pattern_resources.scss:38]: I don't understand this question, no idea what "1005 of the middle 6 columns" means

**Comment by lyubomir-popov** [comment]
@mtruj013 sorry meant 100% of the parent container, which should be a col-6

**Comment by lyubomir-popov** [comment]
looks great thanks @mtruj013 !

**Pull request merged**